pxd.c

```diff
@@ -1513,12 +1513,13 @@ ssize_t pxd_add(struct fuse_conn *fc, struct pxd_add_ext_out *add)
 		}
 	}
 
+	spin_unlock(&ctx->lock);
 	err = pxd_bus_add_dev(pxd_dev);
 	if (err) {
-		spin_unlock(&ctx->lock);
```

Comment:
the above sequence of actions can race between multiple pxd device adds.
one should perform list_add under context lock (L1522+) before releasing and then revert them if failure happens, which is a cleanup op and rarely taken path.

Reply:
Fixed, added device to list before releasing lock. Added cleanup path to remove from list in case of failures.